AngularApp/src/app/modules/dashboard/tabs/tab-data-sources/tab-data-sources.component.ts

```diff
@@ -0,0 +1,54 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { DataProviderMetadata, DetectorResponse } from '../../../../diagnostic-data/models/detector';
+import { ActivatedRoute, Params } from '../../../../../../node_modules/@angular/router';
+import { ApplensDiagnosticService } from '../../services/applens-diagnostic.service';
+
+@Component({
+  selector: 'tab-data-sources',
+  templateUrl: './tab-data-sources.component.html',
+  styleUrls: ['./tab-data-sources.component.css']
+})
+export class TabDataSourcesComponent  {
+
+  constructor(private _route: ActivatedRoute,private _diagnosticApiService: ApplensDiagnosticService) {
+  }
+  
+  detectorResponse: DetectorResponse;
+  detector: string;
+  error: any;
+  loadingDetector : boolean = true;
+
+  ngOnInit() {
+
+    this._route.params.subscribe((params: Params) => {
+      this.getDetectorResponse();
+    });
+
+    this._route.queryParams.subscribe((queryParams: Params) => {
+      this.getDetectorResponse();
+    })
+  }
+
+  refresh() {
```

Comment:
are you using this method? maybe just a copy paste issue..

also, you can consider just inheriting this component code from the Data Component if this is just a copy.
like 
export class Derived extends Base {
}

Reply:
They aren't same code. In this component, this check is different from Data Component...The check for this._route.parent.snapshot.params is not happnening in Parent...

I removed the refresh method which wasnt used...

Reply:
ok..